GameOfLifeApp/Controllers/GameLogic.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+
+namespace GameOfLifeApp 
+{
+    public class GameLogic : IGameLogic
+    {
+        /// <summary>
+        /// Generates a start array of random values according to user`s input.
+        /// </summary>
+        public void SetUpField(IGameField field)
+        {                
+            Random rnd = new Random();
+
+            bool[,] GeneratedField = new bool[field.CountOfRows, field.CountOfColumns];
+
+            for (int row = 0; row < field.CountOfRows; row++)
+            {
+                for (int column = 0; column < field.CountOfColumns; column++)
+                {
+                    GeneratedField[row, column] = rnd.Next(0, 2) == 0;
+                }
+            }
+            field.CurrentField = GeneratedField;
+        }
+
+        /// <summary>
+        /// Populates the field with Glider the Shape. 
+        /// </summary>
+        public void CreateGliderShape(IGameField pr)
+        {
+            pr.CurrentField[0, 0] = true;
+            pr.CurrentField[2, 0] = true;
+            pr.CurrentField[1, 1] = true;
+            pr.CurrentField[2, 1] = true;
+            pr.CurrentField[1, 2] = true;
+        }
+
+        /// <summary>
+        /// Gets number of alive cells around a given cell. 
+        /// </summary>
+        /// <param name="currentRow">Row of a current cell.</param>
+        /// <param name="currentColumn">Column of a current cell.</param>
+        /// <returns>Returns number of alive neighbours.</returns>
+        private int NeighboursCount(IGameField field, int currentRow, int currentColumn)
```

Comment:
you can pass class for the field not the interface.

Reply:
We discussed it in our previous call about 2 weeks ago and decided to leave it at that.